### PR TITLE
fix: prevent children rerender in listwidget

### LINKED ISSUE #11620: [Perf]: List widget re-rendering memoization
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

The `getCanvasChildren` memoization is not working correctly anymore i.e the equality function always returns false. Right now the only way this works is if we do deep equality on all params.

Works
```js
memoizeOne(fn, deepEqual)
```

Doesn't work with individual deep equals
```js
memoizeOne(fn, (prev: any, next: any) => {
      // not comparing canvasChildren becuase template acts as a proxy

      return (
        deepEqual(prev[0], next[0]) &&
        deepEqual(prev[1], next[1]) &&
        deepEqual(prev[2], next[2]) &&
        deepEqual(prev[3] === next[3]) &&
        deepEqual(prev[4] === next[4]) &&
        deepEqual(prev[5] === next[6]) &&
        deepEqual(prev[6] === next[6])
      );
    })
```

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Avoid re-rendering the list widget children in the List widget when making Canvas operations

Fixes #11620

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/ListWidget/widget/index.tsx [hunk 1]
```diff
@@ -749,6 +752,7 @@ class ListWidget extends BaseWidget<ListWidgetProps<WidgetProps>, WidgetState> {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       itemBackgroundColor,
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      perPage,
```
[the remark this answers is not in the document]
Comment (Tooluloope) [reply app/client/src/widgets/ListWidget/widget/index.tsx:755]: Klona is imported in line 46, we can just go with that

[the remark this answers is not in the document]
Comment (SatishGandham) [reply app/client/src/widgets/ListWidget/widget/index.tsx:755]: No. Change that import.

File: app/client/src/widgets/ListWidget/widget/index.tsx [hunk 2]
```diff
@@ -762,21 +767,23 @@ class ListWidget extends BaseWidget<ListWidgetProps<WidgetProps>, WidgetState> {
...
-        prev[4] === next[4] &&
-        prev[5] === next[6] &&
-        prev[6] === next[6]
-      );
-    },
+    (prev: any, next: any) => this.deepEqual(prev, next),
   );
 
+  // deepEqual
+  deepEqual = (prev: any[], next: any[]) => {
```
Comment (SatishGandham) [inline app/client/src/widgets/ListWidget/widget/index.tsx:774]: Sorry, I should have said this earlier. It is a comparison function, not a deepEqual.
Rename it to may be compareProps or something better.

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by SatishGandham — APPROVED** [review]
Looks good. Please create an issue for the PR.

**Comment by somangshu** [comment]
@Tooluloope no issue linked to this, Can you please rectify this

**Pull request merged**